crd/controller.go

```diff
@@ -0,0 +1,179 @@
+package main
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+
+	datasetsv1 "github.com/nerdalize/nerd/crd/pkg/apis/stable.nerdalize.com/v1"
+	clientset "github.com/nerdalize/nerd/crd/pkg/client/clientset/versioned"
+	informers "github.com/nerdalize/nerd/crd/pkg/client/informers/externalversions"
+	listers "github.com/nerdalize/nerd/crd/pkg/client/listers/stable.nerdalize.com/v1"
+)
+
+const (
+	maxRetries = 5
+)
+
+// Controller is the controller implementation for Dataset resources
+type Controller struct {
+	// nerdalizeclientset is a clientset for our own API group
+	nerdalizeclientset clientset.Interface
+	// workqueue is a rate limited work queue. This is used to queue work to be
+	// processed instead of performing it as soon as a change happens. This
+	// means we can ensure we only process a fixed amount of resources at a
+	// time, and makes it easy to ensure we are never processing the same item
+	// simultaneously in two different workers.
+	workqueue      workqueue.RateLimitingInterface
+	informer       cache.SharedIndexInformer
+	datasetsLister listers.DatasetLister
+	eventHandler   Handler
+}
+
+// NewController returns a new sample controller
+func NewController(
```

Comment:
This is awesome work! Hopefully we can use this in the future for more dynamic behaviour